### PR TITLE
fix cmake build

### PR DESCRIPTION
**Pull request by withmorten**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def build(self):
         env = {}
         cmake.definitions["LIBRW_PLATFORM"] = self._librw_platform
         cmake.definitions["LIBRW_INSTALL"] = True
-        cmake.definitions["LIBRW_TOOLS"] = True
+        cmake.definitions["LIBRW_TOOLS"] = False
```
Comment (madebr) [inline conanfile.py:111]:
```suggestion
        cmake.definitions["LIBRW_TOOLS"] = self.options.platform == "null"
```

**Comment by aap** [comment]
I don't think not building the examples/tools is a good solution. it would be a lot better if we do build the examples but such that it actually works. BUT there should be a way to only build core librw because that's all we need for reGTA

**Comment by withmorten** [comment]
@madebr do you have any idea how to achieve that? have the tools in the conanfile enabled for the librw repo, but not in the re3 builds?

also we need to probably add `-std=c++11` to the build flags on mac (at least) for imgui in the tools (the reason it breaks on re3 currently).

**Comment by withmorten** [comment]
gah, now the ubuntu build fails with: `while calling '_build_cmake', line 270
    tools.rename("libpulse.pc", "libpulse-simple.pc")
    ConanException: rename libpulse.pc to libpulse-simple.pc failed, dst exists.`

**Comment by madebr** [comment]
@withmorten 


> @madebr do you have any idea how to achieve that? have the tools in the conanfile enabled for the librw repo, but not in the re3 builds?

This can be done in 2 ways:
- change the default value of [`option`](https://cmake.org/cmake/help/latest/command/option.html) to disable it by default when it is included as a subproject
- use [`cmake_dependent_option`](https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html) to conditionally force disable it

Personally, I prefer the first approach as it is the most flexible.


> also we need to probably add `-std=c++11` to the build flags on mac (at least) for imgui in the tools (the reason it breaks on re3 currently).

This should be fixed in https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html



> gah, now the ubuntu build fails with: `while calling '_build_cmake', line 270 tools.rename("libpulse.pc", "libpulse-simple.pc") ConanException: rename libpulse.pc to libpulse-simple.pc failed, dst exists.`

The problem is that `sdl2/2.0.12@bincrafters/stable` is unmaintained.
There is work to create `sdl/2.0.16`.

For now, we'll have to ignore the sdl failure.

**Comment by withmorten** [comment]
what we currently mostly want is the tools to not be built for re3, but here only, so that the re3 build works again. if you can fix that, please open a PR!

fixing the ci here would be nice, but is not the main priority, this PR is just a testing ground for me right now to see how to achieve that.

**Pull request merged**